test/extended/util/configv1shim_test.go

```diff
@@ -0,0 +1,569 @@
+				{Type: watch.Added, Object: createNetworkObject("staticObject")},
+			},
+		},
+		{
+			name: "merging static and real network objects, field selector no match",
+			staticObjects: []runtime.Object{
+				createNetworkObject("staticObject"),
+			},
+			realObjects: []runtime.Object{
+				createNetworkObject("staticObject"),
+			},
+			watch: func(client *ConfigClientShim, ctx context.Context, opts metav1.ListOptions) (watch.Interface, error) {
+				return client.ConfigV1().Networks().Watch(ctx, opts)
+			},
+			fieldSelector:       "metadata.name=!staticObject",
+			expectedWatchEvents: []watch.Event{},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			configClient := fakeconfigv1client.NewSimpleClientset()
+
+			client := NewConfigClientShim(
+				configClient,
+				test.staticObjects,
+			)
+
+			// The watch request has to created first when using the fake clientset
+			resultChan, err := test.watch(client, context.TODO(), metav1.ListOptions{FieldSelector: test.fieldSelector})
+			if err != nil {
+				t.Fatalf("Expected no error, got %q instead", err)
+			}
+			defer resultChan.Stop()
+
+			// And only then the fake clientset can be populated to generate the watch event
+			for _, obj := range test.realObjects {
+				configClient.Tracker().Add(obj)
+			}
+
+			// verify the watch events
+			ticker := time.NewTicker(500 * time.Millisecond)
+			size := len(test.expectedWatchEvents)
+			eventCounter := 0
+			for i := 0; i < size; i++ {
+				select {
+				case item := <-resultChan.ResultChan():
+					diff := cmp.Diff(test.expectedWatchEvents[i], item)
+					if diff != "" {
+						t.Errorf("test '%s' failed. Results are not deep equal. mismatch (-want +got):\n%s", test.name, diff)
+					}
+					eventCounter++
+				case <-ticker.C:
+					t.Errorf("failed waiting for watch event")
+				}
+			}
+			if eventCounter != size {
+				t.Errorf("Expected %v watch events, got %v instead", eventCounter, size)
+			}
+		})
+	}
+}
+
+func TestConfigClientShimList(t *testing.T) {
+
+	staticObject := createInfrastructureObject("staticObject")
+	staticObject.Labels["static"] = "true"
+
+	realObject1 := createInfrastructureObject("staticObject")
+	realObject1.Labels["static"] = "false"
+
+	realObject2 := createInfrastructureObject("realObject")
+	realObject2.Labels["static"] = "false"
+
+	configClient := fakeconfigv1client.NewSimpleClientset(
+		realObject1,
+		realObject2,
+	)
+
+	client := NewConfigClientShim(
+		configClient,
+		[]runtime.Object{staticObject},
+	)
+
+	listItems, err := client.ConfigV1().Infrastructures().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Expected no error for a List request, got %q instead", err)
+	}
+
+	if len(listItems.Items) != 2 {
+		t.Fatalf("Expected only a single item in the list, got %v instead", len(listItems.Items))
+	}
+
+	var staticObj *configv1.Infrastructure
+
+	for _, item := range listItems.Items {
+		if item.Name == "staticObject" {
```

Comment:
might be good to also check that the 2nd returned object is the real one (`realObject`)

Reply:
Done